backtesting/_plotting.py

```diff
@@ -537,10 +537,24 @@ def __eq__(self, other):
             colors = value._opts['color']
             colors = colors and cycle(_as_list(colors)) or (
                 cycle([next(ohlc_colors)]) if is_overlay else colorgen())
-            legend_label = LegendStr(value.name)
-            for j, arr in enumerate(value, 1):
+
+            tooltip_label = value.name
+
+            if len(value) == 1:
+                assert isinstance(value.name, str)
+                legend_labels = [LegendStr(value.name)]
```

Comment:
This case can be handled by:
```py
legend_labels = [LegendStr(item) for item in _as_list(value.name)]
```
in the branch below. This will then also support use like:
```py
self.I(indicator_1d, name=['foo'])
```
which is reasonable.